massa-protocol-worker/src/handlers/peer_handler/tester.rs

```diff
@@ -234,20 +224,30 @@ impl Tester {
                 }
             };
 
-            // if handshake failed, we set the peer state to HandshakeFailed
-            if res.is_err() {
+            {
                 let mut peer_db_write = peer_db.write();
-                peer_db_write
-                    .peers
-                    .entry(peer_id)
-                    .and_modify(|info| {
-                        info.state = super::PeerState::HandshakeFailed;
-                    })
-                    .or_insert(PeerInfo {
-                        last_announce: None,
-                        state: super::PeerState::HandshakeFailed,
-                    });
+                // if handshake failed, we set the peer state to HandshakeFailed
+                if res.is_err() {
+                    peer_db_write
+                        .peers
+                        .entry(peer_id)
+                        .and_modify(|info| {
+                            info.state = super::PeerState::HandshakeFailed;
+                        })
+                        .or_insert(PeerInfo {
+                            last_announce: None,
+                            state: super::PeerState::HandshakeFailed,
+                        });
+                } else {
+                    let new_md = if let Some(md) = peer_db_write.try_connect_history.remove(&addr) {
```

Comment:
maybe we should separate the attempts made by the testers from the reel connection attempts. 
For the testers, we should prioritize using the tested_addresses map.

Reply:
Here, we only update the metadata associated with this peer, that will be used later for connection attempt

The tester is completely separated from the connection attempts, the only link is the metadata `last_try` that shows when is the last time the peer was  tested for connectivity using the tester.

Reply:
But for me `try_connect` from `connectivity.rs` is different from the connection made by testers and it would be nice to make a difference here.
We have information of when the peer was last tested with `tested_addresses`.
With that we have the last time when the peer was tested by tester and the last time of try_connect was called for the peer.

Reply:
Actually, we don't really care of the last attempt of `try_connect` as we have already the time for the possible outcomes, but I see what you mean, it's overlapping

Reply:
The thing is that I want to have a struct `ConnectionMetadata` containing all the data required to perform priorization, and removing this piece of data would forbid me to implement `PartialOrd` on the structs fields...

I think the best is still to let these overlap a little bit, one is used only for testers (`tested_addresses`), and one only for peer priorization (`try_connect_history`)

Reply:
ok i see. 

You could write like this instead of remove/insert : 
```rust
         peer_db_write
                        .try_connect_history
                        .entry(addr)
                        .and_modify(|md| {
                            md.new_try();
                        })
                        .or_insert(ConnectionMetadata {
                            last_try: Some(MassaTime::now().unwrap()),
                            ..Default::default()
                        });
                        ```
 And why update ConnectionMetadata of the peer only if res.is_ok() ?
 if res_is_err() we should also set new_try to MD, no ?

Reply:
In the metadata:
- `last_success` (updated with `success()`) is when we successfully connected to a peer
- `last_failure` (updated with `failure()`) is when we failed to connect to a peer after it was tested as active
- `last_test` (updated with `test()`) is when we test a peer, whatever the result of this test is

We don't need the time of the last connection attempt as it will be updated in either `last_success` or `last_failure` depending on if it was successful or not

Reply:
Sorry I see what you said now, I'll change this